soci/soci_index.go

```diff
@@ -133,7 +157,14 @@ func WithBuildToolVersion(version string) BuildOption {
 	}
 }
 
-func BuildSociIndex(ctx context.Context, cs content.Store, img images.Image, spanSize int64, store orascontent.Storage, opts ...BuildOption) (*SociIndex, error) {
+func WithCreateORASManifest(val bool) BuildOption {
```

Comment:
Similar to other commens, I would prefer if we took a specific "index type". 

Maybe something like this:
```
type ManifestType string

const (
    ORASManifestType = ManifestType("oras")
    OCIManifestType = ManifestType("oci")
)

func WithManifestType(val ManifestType) BuildOption
```

I'm not totally sold on this exact interface. From the library's perspective, MediaType is the important piece. But requiring a CLI user to type out the full MediaType would be a terrible UX. Maybe there should be a conversion between the CLI name and the real MediaType somewhere in the CLI?

Reply:
I can do something like:

```
manifestType := soci.OCIManifestType
if cliContext.Bool("--create-oras-manifest") {
    manifestType := soci.ORASManifestType
}
BuildSociIndex(..., WithManifestType(manifestType))
```

Reply:
Done.